pandas/tests/series/test_constructors.py

```diff
@@ -725,3 +725,30 @@ def f():
         self.assertEqual(s.dtype, 'timedelta64[ns]')
         s = Series([pd.NaT, np.nan, '1 Day'])
         self.assertEqual(s.dtype, 'timedelta64[ns]')
+
+    def test_constructor_ndarray_name_unhashable(self):
+        for n in [['name_list'], np.ones(2), {1: 2}]:
+            self.assertRaises(TypeError, Series, np.ones(3), name=n)
+
+    def test_constructor_ndarray_name_hashable(self):
+        for n in [777, 777., 'name', datetime(2001, 11, 11), (1, 2)]:
+            s = Series(np.ones(3), name=n)
+            self.assertEqual(s.name, n)
+
+    def test_constructor_dict_name_unhashable(self):
+        for n in [['name_list'], np.ones(2), {1: 2}]:
+            self.assertRaises(TypeError, Series, {'a': 0, 'b': 1}, name=n)
+
+    def test_constructor_dict_name_hashable(self):
+        for n in [777, 777., 'name', datetime(2001, 11, 11), (1, 2)]:
+            s = Series({'a': 0, 'b': 1}, name=n)
```

Comment:
also enhance this test (in ``series/tests_io.py))

```
    def test_pickle_preserve_name(self):
        unpickled = self._pickle_roundtrip_name(self.ts)
        self.assertEqual(unpickled.name, self.ts.name)
```

which loops thru a bunch of possibilities (for the hashables ones). This is to ensure that pickleablity is preserved.